src/micromamba/package.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include "common_options.hpp"
+
+#include "mamba/core/util.hpp"
+#include "mamba/core/package_handling.hpp"
+
+using namespace mamba;  // NOLINT(build/namespaces)
+
+void
+set_package_command(CLI::App* subcom)
+{
+    static std::string infile, dest;
+    static int compression_level = 10;
+
+    auto extract_subcom = subcom->add_subcommand("extract");
+    extract_subcom->add_option("archive", infile, "Archive to extract");
+    extract_subcom->callback([&]() {
+        std::cout << "Extracting " << fs::absolute(infile) << std::endl;
+        extract(fs::absolute(infile));
+    });
+
+    auto compress_subcom = subcom->add_subcommand("compress");
+    compress_subcom->add_option("folder", infile, "Folder to compress");
+    compress_subcom->add_option("dest", dest, "Destination (e.g. myfile-3.1-0.tar.bz2 or .conda)");
+    compress_subcom->add_option("-c,--compression-level",
+                                compression_level,
+                                "Compression level from 0 to 50 (default is 10)");
```

Comment:
Where does that range come from? In the command-line utilities the compression range for bzip2, gzip, zip is 0-9 and the range for zstd is 0-19 IIRC (plus some extra strong compression options that are hidden behind a flag)

Reply:
The library supports regular compression levels from 1 up to ZSTD_maxCLevel(),
  which is currently 22. Levels >= 20, labeled `--ultra`, should be used with
  caution, as they require more memory. The library also offers negative
  compression levels, which extend the range of speed vs. ratio preferences.
  The lower the level, the faster the speed (at the cost of compression).

Reply:
The above is for zstd. I am actually not sure if compression level is implemented for `tar.bz2` :) let me check

Reply:
you're right wrt to bzip2 -- that supports between 0-9 -- I added it as an option.